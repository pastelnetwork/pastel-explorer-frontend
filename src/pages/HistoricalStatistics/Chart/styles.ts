import { makeStyles } from '@material-ui/core/styles';
import { TAppTheme } from '@theme/index';

export const eChartLineStyles = makeStyles((theme: TAppTheme) => ({
  container: {
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    padding: '0 20px',
    textAlign: 'center',
    boxSizing: 'border-box',
    position: 'relative',
    color: 'rgb(171, 170, 193)',
    [theme.breakpoints.down('sm')]: {
      padding: 0,
    },
  },
  lineChartWrap: {
    width: '100%',
    height: 'auto',
    marginBottom: 20,
  },
  lineChartContainer: {
    display: 'block',
    width: '100%',
    height: '100%',
    backgroundColor: '#212121',
  },
  lineChartTitle: {
    width: '100%',
    textAlign: 'left',
    fontSize: '30px',
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
    },
  },
  lineChartHeader: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
    paddingTop: '30px',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      paddingTop: 10,
      paddingLeft: 10,
      justifyContent: 'flex-start',
      flexDirection: 'column',
    },
  },
  uploadButton: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: '#6151d0',
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '14px',
    padding: '8px 16px',
    borderRadius: '4px',
    marginLeft: '8px',
    marginRight: '8px',
    '&:hover': {
      cursor: 'pointer',
      opacity: 0.8,
    },
  },
  reactECharts: {
    width: '100%',
    height: 'calc(100vh - 450px) !important',
  },
  granularitySelect: {
    display: 'flex',
    width: '100%',
    maxWidth: 200,
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 15,
    marginRight: 50,
  },
  periodSelect: {
    display: 'flex',
    minWidth: '200px',
    // justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: '15px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 5,
    },
  },
  lineChartDownloadButtonBar: {
    display: 'flex',
    width: '100%',
    justifyContent: 'flex-end',
    height: '30px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 10,
      justifyContent: 'flex-start',
    },
  },
  lineChartFooter: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingBottom: 30,
    [theme.breakpoints.down('sm')]: {
      flexWrap: 'wrap',
      justifyContent: 'flex-start',
      paddingBottom: 10,
    },
  },
  lineChartThemeSelect: {
    display: 'flex',
    minWidth: '150px',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  themeSelectButton: {
    width: '30px',
    height: '30px',
    borderRadius: '5px',
  },
  filterButton: {
    backgroundColor: '#eee',
    color: '#212124',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '12px',
    padding: '4px 8px',
    borderRadius: '3px',
    borderWidth: '1px',
    borderColor: '#212124',
    marginLeft: '4px',
    marginRight: '4px',
    '&:hover': {
      cursor: 'pointer',
      opacity: '0.8',
    },
  },
  activeButton: {
    backgroundColor: '#c3921f',
    borderColor: '#c3921f',
    color: '#fff',
  },
  activeThemeButton: {
    borderColor: '#008000',
    borderWidth: '5px',
  },
}));
