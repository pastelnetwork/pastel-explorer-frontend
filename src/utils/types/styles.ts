/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/ban-types */
import { SpacingProps } from '@material-ui/system';
import { Theme } from '@material-ui/core';

export type GlobalStyleProps = {
  theme: Theme & { palette: any };
};

export interface MuiButtonSpacingType extends SpacingProps {
  component?: React.PropsWithoutRef<{}>;
  to?: string;
}
