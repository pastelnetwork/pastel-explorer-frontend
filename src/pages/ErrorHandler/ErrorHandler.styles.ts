import styled from 'styled-components/macro';
import { spacing } from '@material-ui/system';
import { Button as MuiButton } from '@material-ui/core';

import { MuiButtonSpacingType } from '@utils/types/styles';

export const Button = styled(MuiButton)<MuiButtonSpacingType>(spacing);

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing(6)}px;
  text-align: center;
  background: transparent;

  ${props => props.theme.breakpoints.up('md')} {
    padding: ${props => props.theme.spacing(10)}px;
  }
`;

export const Logo = styled.img`
  width: 170px;
`;

export const ContentWrapper = styled.div`
  display: flex;
  align-item: center;
  justify-content: center;
  min-height: 85vh;

  .content {
    display: inline-block;
    margin: auto;
  }
`;
