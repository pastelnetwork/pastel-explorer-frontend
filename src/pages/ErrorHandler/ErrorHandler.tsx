import * as React from 'react';
import { Typography, Grid } from '@material-ui/core';
import { Helmet } from 'react-helmet-async';

import { translate } from '@utils/helpers/i18n';
import PastelLogo from '@assets/images/pastel-logo.png';

import * as Styles from './ErrorHandler.styles';

interface IErrorHandlerProps {
  children: React.ReactNode;
}

interface IErrorHandlerState {
  hasError: boolean;
}

class ErrorHandler extends React.Component<IErrorHandlerProps, IErrorHandlerState> {
  state = { hasError: false };

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  componentDidCatch(error: Error) {
    // Integrate here any log error service if needed
    return console.error(error);
  }

  render() {
    const { children } = this.props;
    const { hasError } = this.state;

    if (hasError) {
      return (
        <Styles.Wrapper>
          <Helmet title={translate('pages.errorHandler.pageError')} />
          <Styles.ContentWrapper>
            <div className="content">
              <Grid container justify="center" alignItems="center" direction="column" spacing={5}>
                <Grid item>
                  <Styles.Logo src={PastelLogo} alt={translate('common.pastelLogo')} />
                </Grid>
                <Grid item>
                  <Typography component="h1" variant="h1" align="center" gutterBottom>
                    {translate('pages.errorHandler.generalError')}
                  </Typography>
                </Grid>
              </Grid>
              <Typography component="h2" variant="h5" align="center" gutterBottom>
                {translate('pages.errorHandler.description')}
              </Typography>
              <Typography component="h2" variant="body1" align="center" gutterBottom>
                {translate('pages.errorHandler.refresh')}
              </Typography>

              <Styles.Button
                onClick={() => window.location.reload()}
                variant="contained"
                color="secondary"
                mt={2}
              >
                {translate('pages.errorHandler.btnRefresh')}
              </Styles.Button>
            </div>
          </Styles.ContentWrapper>
        </Styles.Wrapper>
      );
    }

    return children;
  }
}

export default ErrorHandler;
