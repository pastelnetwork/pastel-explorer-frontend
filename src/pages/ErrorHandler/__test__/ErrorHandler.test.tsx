import { shallow } from 'enzyme';
import { Typography, Grid } from '@material-ui/core';
import 'jest-styled-components';

import i18next from '../../../utils/helpers/i18n';
import ErrorHandler from '../ErrorHandler';
import * as Styles from '../ErrorHandler.styles';

jest.mock('i18next-http-backend');
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
});

describe('pages/ErrorHandler', () => {
  const wrapper = shallow(
    <ErrorHandler>
      <div>Content</div>
    </ErrorHandler>,
  );

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <Grid>', () => {
    expect(wrapper.find(Grid).length).toBe(0);
  });

  test('should render <Typography>', () => {
    expect(wrapper.find(Typography).length).toBe(0);
  });

  test('should render <Styles.Wrapper>', () => {
    expect(wrapper.find(Styles.Wrapper).length).toBe(0);
  });

  test('should render <Styles.ContentWrapper>', () => {
    expect(wrapper.find(Styles.ContentWrapper).length).toBe(0);
  });

  test('should render <Styles.Button>', () => {
    expect(wrapper.find(Styles.Button).length).toBe(0);
  });
});
