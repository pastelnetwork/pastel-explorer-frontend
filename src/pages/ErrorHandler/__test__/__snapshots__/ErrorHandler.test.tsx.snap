// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/ErrorHandler renders correctly 1`] = `ShallowWrapper {}`;
